Source/JavaScriptCore/assembler/LinkBuffer.cpp

```diff
@@ -472,6 +473,48 @@ void LinkBuffer::linkCode(MacroAssembler& macroAssembler, JITCompilationEffort e
     macroAssembler.padBeforePatch();
 
 #if ENABLE(JIT)
+#if ENABLE(JIT_SCAN_ASSEMBLER_BUFFER_FOR_ZEROES)
+    {
+        // On x86-64, the maximum immediate size is 8B, no opcodes/prefixes have 0x00
+        // On other architectures this could be smaller
+        constexpr size_t maxZeroByteRunLength = 16;
+        // This algorithm works because the number of 0-bytes which can fit into
+        // one qword (8) is smaller than the limit on which we assert (9).
+        constexpr size_t stride = sizeof(uint64_t);
+        static_assert(stride <= maxZeroByteRunLength);
+
+        const AssemblerBuffer& buffer = macroAssembler.m_assembler.buffer();
+        size_t codeSize = buffer.codeSize();
+        const char* code = reinterpret_cast<const char*>(buffer.constData());
+        size_t runLength = 0;
+        size_t i = 0;
+        if (codeSize > stride) {
+            for (; i % stride; i++) {
```

Comment:
If `i` is always initialized to zero, is this loop necessary? `i % stride` would end up as `0` immediately, meaning that we'd just exit the loop (unless I'm missing something?).